rest_framework/validators.py

```diff
@@ -188,7 +188,10 @@ def __call__(self, attrs, serializer):
                 if attrs[field_name] != getattr(serializer.instance, field_name)
             ]
 
-        condition_kwargs = {source: attrs[source] for source in self.condition_fields}
+        condition_kwargs = {
```

Comment:
the change and the test seems OK to be from the first glance, however, I would love to see, if this change helps to fix more test cases or not. For that reason, would you mind checking the failing test cases here https://github.com/encode/django-rest-framework/pull/9678 ? and try to incorporate relevant parts in this PR so re check?

Reply:
Yeah, I saw that and figured this issue was at least tangentially related. I wanted to keep this change focused to, well to be honest, make it easy on myself. 😄 I'll take a deeper look and see if there's any overlap.

Reply:
thanks

Reply:
Yes it's a good question... 

My impression was that they were likely separate. I was therefore going to suggest pulling `master` into the other branch after we merge this. In the best case, we get a green build of the other branch and can merge the added test.

Testing ahead of merging is a good idea, although I would lean on the side of leaving them separate if this doesn't fix the other test

Reply:
yes that also make sense...

Reply:
I added the tests from #9678 and they still don't pass